File: app/src/main/java/ru/samtakoy/core/presentation/qpack/CardViewingTypeSelector.java
package ru.samtakoy.core.presentation.qpack;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import ru.samtakoy.R;
import ru.samtakoy.core.presentation.log.MyLog;

public class CardViewingTypeSelector extends DialogFragment {


    public static DialogFragment newFragment(QPackInfoFragment targetFragment, int requestCode) {
        CardViewingTypeSelector result = new CardViewingTypeSelector();
        result.setCancelable(true);
        result.setTargetFragment(targetFragment, requestCode);
        return result;
    }

    enum CardViewingType{
        RANDOM,
        SIMPLE,
        LIST
    };

    interface CardViewingTypeSelectorListener{
        void OnCardViewingTypeSelect(CardViewingType type);
    }

    private int mPositionIndex = 0;


    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(!(getTargetFragment() instanceof CardViewingTypeSelectorListener))
        {
            MyLog.add("CardViewingTypeSelectorListener must be implemented in target fragment");
            //throw new ClassCastException("CardViewingTypeSelectorListener must be implemented in target fragment");
            dismiss();
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        String[] items = getContext().getResources().getStringArray(R.array.qpack_viewing_types);

        AlertDialog.Builder builder = new AlertDialog.Builder(getContext())
                .setTitle(R.string.qpack_viewing_types_title)
                .setSingleChoiceItems(items, mPositionIndex, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        mPositionIndex = i;
                    }
                })
                .setPositiveButton(R.string.btn_ok, (dialogInterface, i) -> onOk())
                .setNegativeButton(R.string.btn_cancel, (dialogInterface, i) -> dismiss())
                ;
        return builder.create();
    }

    private void onOk(){
        CardViewingTypeSelectorListener listener = (CardViewingTypeSelectorListener)getTargetFragment();
        listener.OnCardViewingTypeSelect(CardViewingType.values()[mPositionIndex]);
        dismiss();
    }

}

File: app/src/main/kotlin/ru/samtakoy/core/presentation/qpack/info/CardViewingTypeSelector.kt
package ru.samtakoy.core.presentation.qpack.info

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import ru.samtakoy.R
import ru.samtakoy.common.utils.MyLog.add

class CardViewingTypeSelector : DialogFragment() {
    enum class CardViewingType {
        RANDOM,
        SIMPLE,
        LIST
    }

    internal interface CardViewingTypeSelectorListener {
        fun OnCardViewingTypeSelect(type: CardViewingType?)
    }

    private var mPositionIndex = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (getTargetFragment() !is CardViewingTypeSelectorListener) {
            add("CardViewingTypeSelectorListener must be implemented in target fragment")
            //throw new ClassCastException("CardViewingTypeSelectorListener must be implemented in target fragment");
            dismiss()
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val items = requireContext().getResources().getStringArray(R.array.qpack_viewing_types)

        val builder = AlertDialog.Builder(requireContext())
            .setTitle(R.string.qpack_viewing_types_title)
            .setSingleChoiceItems(items, mPositionIndex, object : DialogInterface.OnClickListener {
                override fun onClick(dialogInterface: DialogInterface?, i: Int) {
                    mPositionIndex = i
                }
            })
            .setPositiveButton(
                R.string.btn_ok,
                DialogInterface.OnClickListener { dialogInterface: DialogInterface?, i: Int -> onOk() })
            .setNegativeButton(
                R.string.btn_cancel,
                DialogInterface.OnClickListener { dialogInterface: DialogInterface?, i: Int -> dismiss() })

        return builder.create()
    }

    private fun onOk() {
        val listener = getTargetFragment() as CardViewingTypeSelectorListener?
        listener!!.OnCardViewingTypeSelect(CardViewingType.entries[mPositionIndex])
        dismiss()
    }

    companion object {
        fun newFragment(targetFragment: QPackInfoFragment?, requestCode: Int): DialogFragment {
            val result = CardViewingTypeSelector()
            result.setCancelable(true)
            result.setTargetFragment(targetFragment, requestCode)
            return result
        }
    }
}
